File: nitpickers/pyql/pyqls/ast/nodes/page.py
from util.ast import ASTNode


class Page(ASTNode):

    def __init__(self, location, identifier, block):
        super().__init__(location)
        self._identifier = identifier
        self._block = block

    @property
    def identifier(self):
        return self._identifier

    @property
    def block(self):
        return self._block

    def __repr__(self):
        return "Page at {0}".format(self.location)

    __str__ = __repr__

File: GunOli/src/main/java/QLS/AST/Page.java
package QLS.AST;

import QLS.Analysis.WidgetVisitorInterface;
import QLS.AST.Statements.Statement;

import java.util.ArrayList;

public class Page extends QLSNode {

    private ArrayList<Statement> statements;
    private String identifier;

    public Page(ArrayList<Statement> statements, String identifier, int line){
        super(line);
        this.statements = statements;
        this.identifier = identifier;
    }

    public ArrayList<Statement> getStatements() {
        return statements;
    }


    public String getIdentifier() {
        return identifier;
    }

    @Override
    public <T> T accept(WidgetVisitorInterface<T> visitor) {
        return visitor.visit(this);
    }
}
